pkg/storage/engine.go

```diff
@@ -539,28 +539,25 @@ type Writer interface {
 	//
 	// It is safe to modify the contents of the arguments after it returns.
 	ClearRawRange(start, end roachpb.Key) error
-	// ClearMVCCRangeAndIntents removes MVCC keys and intents from start (inclusive)
-	// to end (exclusive). This is a higher-level method that handles both
-	// interleaved and separated intents. Similar to the other Clear* methods,
-	// this method actually removes entries from the storage engine.
+	// ClearMVCCRange removes MVCC keys from start (inclusive) to end (exclusive),
+	// including intents. Similar to the other Clear* methods, this method
+	// actually removes entries from the storage engine.
 	//
 	// It is safe to modify the contents of the arguments after it returns.
-	ClearMVCCRangeAndIntents(start, end roachpb.Key) error
-	// ClearMVCCRange removes MVCC keys from start (inclusive) to end
-	// (exclusive). It should not be expected to clear intents, though may clear
-	// interleaved intents that it encounters. It is meant for efficiently
+	ClearMVCCRange(start, end roachpb.Key) error
+	// ClearMVCCVersions removes MVCC key versions from start (inclusive) to end
+	// (exclusive). It does not affect intents. It is meant for efficiently
 	// clearing a subset of versions of a key, since the parameters are MVCCKeys
 	// and not roachpb.Keys. Similar to the other Clear* methods, this method
```

Comment:
IMHO we need to be more specific here. Method is meant to remove versions of a key in mvcc space, but what if it spans multiple roachpb.Key? I assume it would just remove everything ignoring intents inbetween or maybe it will raise an error about intent. I think we should either say what happens or mention that crossing key boundaries produces undefined behaviour or something in this lines.

Reply:
Good call -- this can be used across multiple keys too, clarified the comment.